### DISCUSSION TITLE
How to conditionally enable / disable detail render per row

### DISCUSSION THREAD

**Discussion by wmoore8** [KevinVandy/material-react-table#593] [General]
Hello!

In the detail render of each row of my grid, some of the rows have content to show while others have none. Is there a way to conditionally enable or disable the detail render (button and all) of each row on a case by case basis, depending on the existence of content within it?

**Comment by KevinVandy** [comment]
This is how my storybook stories do it https://www.material-react-table.dev/?path=/story/features-detail-panel-examples--detail-panel-enabled-conditional

```tsx
muiExpandButtonProps={({ row }) => ({
  sx: {
    display: row.original.age > 50 ? 'flex' : 'none'
  }
})}
```

**Reply by wmoore8** [reply]
Oh amazing! That worked great, thanks so much
